resources.go

```diff
@@ -884,7 +912,8 @@ func Provider() tfbridge.ProviderInfo {
 						AltTypes: []tokens.Type{awsTypeNoFile(iamMod, "InstanceProfile")},
 					},
 					"instance_type": {
-						Type: awsTypeDefaultFile(ec2Mod, "InstanceType"),
+						Type:     "string",
+						AltTypes: []tokens.Type{awsTypeDefaultFile(ec2Mod, "InstanceType")},
```

Comment:
At least to date I believe our policy has been to intentionally use enums to narrow string types.  If we make the change here, the enum really has no value - it may as well just be `string`.

`string | "t2.micro"` is identical to `string`.  You won't get any type-checking, intellisense, etc. benefits.

That said, keeping this as is does have significant downsides if we don't have a robust way to keep these up to date, and honestly probably does cause more friction in practice than it's worth.

We should make a decision on this point separately from the other changes in this PR.  We should also make sure to write this decision down in resource modeling design guidelines.

cc @CyrusNajmabadi @pgavlin @EvanBoyle for input.

Reply:
More background on this in https://github.com/pulumi/pulumi-aws/pull/739.

Reply:
I don't have context around how these values flow through later on and what control we have over manipulation, but could we still get autocomplete if we did something along the lines off:

```ts
type  InstanceOverride = {
  override: string
}
type InstanceType = "t1micro" | "t2micro"; // ..... etc. 

type Instance = InstanceType | InstanceOverride; 

```

The goal would be to keep instanceType as up to date as possible for autocomplete, and to have instanceOverride as an escape hatch that doesn't interfere with string auotocompletion. Not sure if this would even work?